src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/InputRightCurlyAloneOrSinglelineLambda.java

```diff
@@ -0,0 +1,49 @@
+package com.puppycrawl.tools.checkstyle.checks.blocks;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class InputRightCurlyAloneOrSinglelineLambda {
+
+    static Runnable r1 = () -> {
+        String.valueOf("Test rightCurly one!");
+    };
+
+    static Runnable r2 = () -> String.valueOf("Test rightCurly two!");
+
+    static Runnable r3 = () -> {String.valueOf("Test rightCurly three!");};
+
+    static Runnable r4 = () -> {
+        String.valueOf("Test rightCurly four!");};    //violation
+
+    static Runnable r5 = () ->
+    {
+        String.valueOf("Test rightCurly five!");
+    };
+
+    static Runnable r6 = () -> {};
+
+    static Runnable r7 = () -> {
+    };
+
+    static Runnable r8 = () ->
+    {
+    };
+
+    static Runnable r9 = () -> {
+        String.valueOf("Test rightCurly nine!");
+    }; int i;       // violation
+
+    void foo1() {
+        Stream.of("Hello").filter(s -> {
+                return s != null;
+            }
+        ).collect(Collectors.toList());
+
+        Stream.of("Hello").filter(s -> {
+                return s != null;
+        }).collect(Collectors.toList());    // violation
+
+        Stream.of("Hello").filter(s -> {return s != null;}).collect(Collectors.toList());   // violation
```

Comment:
This is how check behaves currently, check out this [example](http://rveach.no-ip.org/checkstyle/checkstyle.php?action=view&config=230681bd222b19e3f97ffdc5c4fd002c&code=34a35d325d4bf900a839f3b4adad317d&checkstyle=checkstyle-7.6.1-all.jar&printTree=).